src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.TestHelpers.fs

```diff
@@ -0,0 +1,187 @@
+open Microsoft.Build.Logging.StructuredLogger
+open CliWrap
+open System
+open System.IO
+open FSharp.Compiler.CodeAnalysis
+
+type DotNetVersion =
+    | Six
+    | Seven
+
+    override this.ToString() =
+        match this with
+        | Six -> "net6.0"
+        | Seven -> "net7.0"
+
+type FSharpProjectOptions with
+
+    static member zero =
+        {
+            ProjectFileName = ""
+            ProjectId = None
+            SourceFiles = [||]
+            OtherOptions = [||]
+            ReferencedProjects = [||]
+            IsIncompleteTypeCheckEnvironment = false
+            UseScriptResolutionRules = false
+            LoadTime = DateTime.UtcNow
+            UnresolvedReferences = None
+            OriginalLoadReferences = []
+            Stamp = None
+        }
+
+type Package = { Name: string; Version: string }
+
+let fsharpFiles = set [| ".fs"; ".fsi"; ".fsx" |]
+
+let isFSharpFile (file: string) =
+    Seq.exists (fun (ext: string) -> file.EndsWith ext) fsharpFiles
+
+let readCompilerArgsFromBinLog file =
+    let build = BinaryLog.ReadBuild file
+
+    let projectName =
+        build.Children
+        |> Seq.choose (
+            function
+            | :? Project as p -> Some p.Name
+            | _ -> None
+        )
+        |> Seq.distinct
+        |> Seq.exactlyOne
+
+    let message (fscTask: FscTask) =
+        fscTask.Children
+        |> Seq.tryPick (
+            function
+            | :? Message as m when m.Text.Contains "fsc" -> Some m.Text
+            | _ -> None
+        )
+
+    let mutable args = None
+
+    build.VisitAllChildren<Task>(fun task ->
+        match task with
+        | :? FscTask as fscTask ->
+            match fscTask.Parent.Parent with
+            | :? Project as p when p.Name = projectName -> args <- message fscTask
+            | _ -> ()
+        | _ -> ()
+    )
+
+    match args with
+    | None -> failwith $"Could not parse binlog at {file}, does it contain CoreCompile?"
+    | Some args ->
+        let idx = args.IndexOf "-o:"
+        args.Substring(idx).Split [| '\n' |]
+
+let mkOptions (compilerArgs: string array) =
+    let sourceFiles =
+        compilerArgs
+        |> Array.filter (fun (line: string) -> isFSharpFile line && File.Exists line)
+
+    let otherOptions =
+        compilerArgs |> Array.filter (fun line -> not (isFSharpFile line))
+
+    {
+        ProjectFileName = "Project"
+        ProjectId = None
+        SourceFiles = sourceFiles
+        OtherOptions = otherOptions
+        ReferencedProjects = [||]
+        IsIncompleteTypeCheckEnvironment = false
+        UseScriptResolutionRules = false
+        LoadTime = DateTime.Now
+        UnresolvedReferences = None
+        OriginalLoadReferences = []
+        Stamp = None
+    }
+
+let mkOptionsFromBinaryLog binLogPath =
+    let compilerArgs = readCompilerArgsFromBinLog binLogPath
+    mkOptions compilerArgs
+
+let mkOptionsFromProject (version: DotNetVersion) (additionalPkg: Package option) =
```

Comment:
This needs a `try/with`, so much stuff that could go wrong.
I also think we need to reconsider how we make the binlog file.
If I run my tests twice I don't want to have two binlog files that carry the same information.
The second run should reuse the evaluation of the first run.

Proposal: Create a unique file name based on the input parameters. Inside the temp folder, we create an `FSharp.Analyzers.SDK` cache folder (where we store all the bin logs).
If the file doesn't exist yet, we do the `dotnet new classlib` dance, otherwise, we read the binlog again. The SDK version should also be included in the unique file name.

For example, something along the lines of:
`mkOptionsFromProject "net6.0" [ { Name = "Fantomas.Core"; Version = "6.2" } ]` would be
`v0.13.0_net6.0_Fantomas.Core_6.2`